src/hooks/useNetworkEnsAddress.ts

```
@@ -0,0 +1,20 @@
+import { mainnet } from 'viem/chains';
+import { useEnsAddress } from 'wagmi';
+import {
+  supportedEnsNetworks,
+  useNetworkConfigStore,
+} from '../providers/NetworkConfig/useNetworkConfigStore';
+
+interface UseNetworkEnsAddressProps {
+  name?: string;
+  chainId?: number;
+}
+
+export function useNetworkEnsAddress(props: UseNetworkEnsAddressProps | undefined) {
+  const { chain } = useNetworkConfigStore();
+  const ensNetworkOrMainnet = supportedEnsNetworks.includes(props?.chainId ?? chain.id)
+    ? chain.id
+    : mainnet.id;
```

Hm, just want to make sure I understand this ternary here.

then a call to this function might look like this:
- props.chainId = mainnet
- chain.id (from newtork config store hook) = sepolia

but then we'd be targeting sepolia in the wagmi hook call, despite passing in mainnet, to the outer hook's props. is that right?

---

o opps I see my mistake. 

Also I think instead of this safe guard I should throw an error. This isn't user entered and would be a mistake on our side if this isn't right.

---

updated.

---

Well your comment made me rethink this. 'We' did something wrong if the network used in this hook is incorrect. so its a runtime safety for us. I could type safety instead?